src/AppBundle/Controller/AdminStatsController.php

```diff
@@ -1,103 +1,264 @@
+                [
+                    'project'=>$projectQuery,
+                    'start' => $startDate,
+                    'end' => $endDate,
+                ]
+            );
         } elseif ($projectQuery != "" && $endDate != "") {
-            return $this->redirectToRoute("AdminStatsResult", [
-                'project'=>$projectQuery,
-                'start' => "1970-01-01",
-                'end' => $endDate,
-            ]);
+            // Redirect if we have the enddate and project
+            return $this->redirectToRoute(
+                "AdminStatsResult",
+                [
+                    'project'=>$projectQuery,
+                    'end' => $endDate,
+                ]
+            );
         } elseif ($projectQuery != "" && $startDate != "") {
-            return $this->redirectToRoute("AdminStatsResult", [
-                'project' => $projectQuery,
-                'start' => $startDate,
-            ]);
+            // Redirect if we have the stardate and project
+            return $this->redirectToRoute(
+                "AdminStatsResult",
+                [
+                    'project' => $projectQuery,
+                    'start' => $startDate,
+                ]
+            );
         } elseif ($projectQuery != "") {
-            return $this->redirectToRoute("AdminStatsResult", [ 'project'=>$projectQuery ]);
+            // Redirect if we have the project name
+            return $this->redirectToRoute(
+                "AdminStatsResult",
+                [
+                    'project'=>$projectQuery
+                ]
+            );
         }
 
         // Otherwise fall through.
-        return $this->render('adminStats/index.html.twig', [
-            'base_dir' => realpath($this->getParameter('kernel.root_dir').'/..'),
-            "xtPage" => "adminstats",
-            "xtPageTitle" => "tool_adminstats",
-            "xtSubtitle" => "tool_adminstats_desc",
-        ]);
+        return $this->render(
+            'adminStats/index.html.twig',
+            [
+                'base_dir' => realpath($this->getParameter('kernel.root_dir').'/..'),
+                "xtPage" => "adminstats",
+                "xtPageTitle" => "tool_adminstats",
+                "xtSubtitle" => "tool_adminstats_desc",
+            ]
+        );
     }
 
-    // TODO: Handle start and end date
     /**
+     * Method for rendering the AdminStats Results
+     *
+     * @param string $project Project to run the results against
+     * @param string $start   Date to start on.  Must parse by strtotime.
+     * @param string $end     Date to end on.  Must parse by strtotime.
+     *
      * @Route("/adminstats/{project}/{start}/{end}", name="AdminStatsResult")
+     *
+     * @return Route|\Symfony\Component\HttpFoundation\Response
      */
-    public function resultAction($project, $start = null, $end = null)
+    public function resultAction($project, $start = "1970-1-1", $end = "2038-1-18")
```

Comment:
Is this making the default query be "for all time"? Because it seems to be that on my local install, which is making it go dreadfully slow.